### DISCUSSION TITLE
当设置接口 结构体 时间类型为 DateTime 时。接口返回（例如：2023-12-21T16:08:48Z），在前端做时间处理时异常

### DISCUSSION THREAD

**Discussion by tpc-ht** [rbatis/rbatis#482] [Ideas]
![image](https://github.com/rbatis/rbatis/assets/59469126/291fcb85-1308-4834-abbf-38e6c4e12e5e)
![image](https://github.com/rbatis/rbatis/assets/59469126/b423c84c-f95d-4480-8e29-da255602753a)
服务端调用 .format 解析是对的。但查询sql与响应的 结构体 都是一个所以并未做二次格式化处理，想直接前端处理。

请问有什么办法可以解决了

**Comment by zhuxiujia** [comment]
> ![image](https://private-user-images.githubusercontent.com/59469126/296162578-291fcb85-1308-4834-abbf-38e6c4e12e5e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyMjAwNjEsIm5iZiI6MTcwNTIxOTc2MSwicGF0aCI6Ii81OTQ2OTEyNi8yOTYxNjI1NzgtMjkxZmNiODUtMTMwOC00ODM0LWFiYmYtMzhlNmM0ZTEyZTVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDA4MDkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhNzRkNGYxMmM5MWM2MjFkOTgwMTJhZDQ4MDE4MDM2MTIzN2VhMzc4ODEyZWNiMzc2MGFiODk1Y2ZiZTE0MGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IFixYaH3k07EQtPIG30h6vxgG-9206XGtFcOXuCU-fE) ![image](https://private-user-images.githubusercontent.com/59469126/296165922-b423c84c-f95d-4480-8e29-da255602753a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyMjAwNjEsIm5iZiI6MTcwNTIxOTc2MSwicGF0aCI6Ii81OTQ2OTEyNi8yOTYxNjU5MjItYjQyM2M4NGMtZjk1ZC00NDgwLThlMjktZGEyNTU2MDI3NTNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDA4MDkyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxNTlkMDQwOGY0ZjU1YmI1N2Q4ZWYyZWFhYmMyZmMzNjUyZWI3MTVhYzZiZjdiNzI1ZDhhNWJkMzAzYzg3NWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YKLmp8fMC4k6SV_ES26t7mJs1PtR-JVcTuqdsdiPjn8) 服务端调用 .format 解析是对的。但查询sql与响应的 结构体 都是一个所以并未做二次格式化处理，想直接前端处理。
> 
> 请问有什么办法可以解决了

可选方案
*  前端修改一下代码支持RFC3339时间编码即“2023-12-12T:00:00:00Z”/“2023-12-12T:00:00:00+08:00”
*  服务端增加一层VO层（MVC设计模式的View层），例如
```rust
UserVO{  pub time:String  }

impl From<User> for UserVO{
   .....
}

//业务代码
#[tokio::main]
async fn main(){
  let user:User = User::select_by_column(rb,"id",1).await.unwrap();
  let user_vo = UserVO::from(user);
 // 返回 user_vo  给前端
}
```

**Reply by tpc-ht** [reply]
谢谢大佬，这个写法很好用
